### PR TITLE
plugins: Fetch the plugin sources from flake inputs

### PR DESCRIPTION
**Pull request by traxys**
Using flake inputs is easier to manage than hardcoding the rev/hash in the plugins def file

### CODE REVIEW

**Review by GaetanLepage — APPROVED** [review]
LGTM !

**Comment by traxys** [comment]
@pta2002 if you could look at this too, I'm not sure if you tried to do this in the past and hit an hurdle somewhere that I missed

**Comment by GaetanLepage** [comment]
Works fine for me (using home-manager).

**Comment by pta2002** [comment]
I have two conflicting opinions on this:

1. This makes managing the external plugins a lot easier, since it's all in the main flake, but...
2. It fills up our flake.nix with things that are better handled elsewhere, and ends up with a ton of inputs for people who might not want to use this

Two possible solutions:
1. Create a new file, and dynamically populate the flake's inputs (something like `inputs = import plugins/external.nix // {...}`
2. Do something like nixpkgs does, with a simple list and then a script that generates the inputs using something like `nix-prefetch-git`

**Comment by pta2002** [comment]
I recon something like `import` should still work, no? It's a matter of trying. Still, I'd be more in favor of the nvfetcher approach

**Comment by traxys** [comment]
When doing something like this you get the following error: `
error: expected a set but got a thunk at /nix/store/8r4pmhh4szpm4i7pabjmld4g1gpkjg8a-source/flake.nix:3:3`

**Comment by traxys** [comment]
@pta2002 Do you like this approach better ?

File: flake.nix [hunk 1]
```diff
@@ -124,7 +127,34 @@
...
+                dontUnpack = true;
+                dontBuild = true;
+                installPhase = let
+                  script = writeShellScript "alejandra-excludes.sh" ''
+                    ${alejandra}/bin/alejandra --exclude ./plugins/_sources "$@"
+                  '';
+                in ''
+                  mkdir -p $out/bin
+                  cp ${script} $out/bin/alejandra-excludes
+                  cat $out/bin/alejandra-excludes
```
Comment (GaetanLepage) [inline flake.nix:153]: Why do you `cat` the script content here ?

Comment (traxys) [reply flake.nix:153]: Oops forgot to remove after debugging

**Comment by pta2002** [comment]
I like it, is there any way to add `nvfetch` to one of the `nix run` options?

**Comment by traxys** [comment]
Yeah I can look into that

**Pull request merged**